applications/CompressiblePotentialFlowApplication/python_scripts/potential_flow_solver.py

```diff
@@ -68,9 +105,15 @@ def AddVariables(self):
         # Degrees of freedom
         self.main_model_part.AddNodalSolutionStepVariable(KCPFApp.VELOCITY_POTENTIAL)
         self.main_model_part.AddNodalSolutionStepVariable(KCPFApp.AUXILIARY_VELOCITY_POTENTIAL)
-
+        # Embedded variables
```

Comment:
Yes, the GEOMETRY_DISTANCE is only for the embedded case!

The rest of variables are defined as historical since they are either used or required in/by Kratos processes. 

DISTANCE, DISTANCE_GRADIENT in calculate_distance_to_skin_process, metrics_level_set_process
FLAG_VARIABLE in variational_distance_process
NODAL_H, VELOCITY, PRESSURE in modify_distance_process (see #4834)

Reply:
They shouldn't, but right now the level-set calculation processes assume that the `DISTANCE` variables are historical. Same happens for the distance modification ones, which require historical `NODAL_H`, `VELOCITY` and `PRESSURE`. I think that @marcnunezc already opened an issue complaining about that.